### PR TITLE
[native_assets_cli] Improve documentation for `DynamicLoadingBundled`

### PR DESCRIPTION
**Pull request by dcharkes**
Closes: https://github.com/dart-lang/native/issues/1049

### CODE REVIEW

File: pkgs/native_assets_cli/lib/src/api/native_code_asset.dart [hunk 1]
```diff
@@ -64,8 +64,9 @@ abstract final class NativeCodeAsset implements Asset {
   /// How this file is bundled depends on the kind of asset, represented by a
   /// concrete subtype of [Asset], and the SDK (Dart or Flutter).
   ///
-  /// If the [linkMode] is [DynamicLoadingBundled], the file most be provided in
-  /// the [BuildOutput] for [BuildConfig.dryRun].
+  /// If the [linkMode] is [DynamicLoadingBundled], the file must be provided in
```
Comment (mosuem) [inline pkgs/native_assets_cli/lib/src/api/native_code_asset.dart:67]: This paragraph is a bit confusing - you must provide a file for a dry run, but actually a file name is enough?

**Review by mosuem — APPROVED** [review]

File: pkgs/native_assets_cli/lib/src/api/native_code_asset.dart (hunk 1 above)
Comment (dcharkes) [reply pkgs/native_assets_cli/lib/src/api/native_code_asset.dart:67]: Yeah. As the full path doesn't really make sense in dry run, the full path is going to be different for the 2 or 3 wet runs (for the different architectures) that follow later in the flutter build.

Suggestions on how to clarify this?

Comment (mosuem) [reply pkgs/native_assets_cli/lib/src/api/native_code_asset.dart:67]: But I thought dry runs actually do expect a real file, so the path must also be real? Why would it change?

Comment (dcharkes) [reply pkgs/native_assets_cli/lib/src/api/native_code_asset.dart:67]: No, dry runs only need to provide the name, not the file.

Comment (mosuem) [reply pkgs/native_assets_cli/lib/src/api/native_code_asset.dart]: Then I would insert a `not`, or what am I missing?
```
/// If the [linkMode] is [DynamicLoadingBundled], the file must not be provided
/// in the [BuildOutput] for [BuildConfig.dryRun]. Supplying a file name instead 
/// of an absolute path is enough for [BuildConfig.dryRun].
```

Comment (mosuem) [reply pkgs/native_assets_cli/lib/src/api/native_code_asset.dart]: Or is this a confusion between the file existing, and the Dart `File` object existing?

Comment (dcharkes) [reply pkgs/native_assets_cli/lib/src/api/native_code_asset.dart]: Indeed, the file on disk does not have to exist in dry run, but the `Url file` must be provided. 🙈

**Review by mosuem — APPROVED** [review]

**Pull request merged**